ui/components/FormDialog.vue

```diff
@@ -0,0 +1,155 @@
+<template>
+  <simple-dialog
+    v-model="value"
+    class="form-dialog"
+    :class="`form-dialog_${mode}`"
+    :actions="actions"
+    :back-disabled="isFirstTab"
+    :details-route="detailsRoute"
+    :height="height"
+    :is-valid="isTabValid"
+    :on-submit="submit"
+    :title="computedTitle"
+    :width="width"
+    @back="goToPreviousTab"
+    @next="goToNextTab"
+  >
+    <template #sidebar>
+      <vertical-tabs
+        v-model:active-tab-key="activeTabKey"
+        :tabs="tabs"
+        :linear="mode === 'wizard'"
+      />
+    </template>
+    <template #default>
+      <slot :name="activeTabKey" />
+    </template>
+  </simple-dialog>
+</template>
+
+<script setup>
+import { ref, computed, watch } from 'vue';
+
+import SimpleDialog from '~/components/SimpleDialog.vue';
+import VerticalTabs from '~/components/VerticalTabs.vue';
+
+const props = defineProps({
+  title: {
+    type: String,
+    required: true,
+  },
+  tabs: {
+    type: Array,
+    required: true,
+  },
+  mode: {
+    type: String,
+    required: true,
+    validator(value) {
+      return ['wizard', 'edit'].includes(value);
```

Comment:
could be a const ['wizard', 'edit']

Reply:
Sadly Vue does not support to use external variables in these compiler macros (defineProps, defineModel, defineEmits...) :/